subprojects/core-api/src/main/java/org/gradle/api/artifacts/repositories/UrlArtifactRepository.java

```diff
@@ -27,24 +30,9 @@ public interface UrlArtifactRepository {
 
     /**
      * The base URL of this repository.
-     *
-     * @return The URL.
-     */
-    URI getUrl();
-
-    /**
-     * Sets the base URL of this repository.
-     *
-     * @param url The base URL.
      */
-    void setUrl(URI url);
-
-    /**
-     * Sets the base URL of this repository.
-     *
-     * @param url The base URL.
-     */
-    void setUrl(Object url);
+    @ReplacesEagerProperty
+    Property<URI> getUrl();
```

Comment:
❓ Why we don't need to upgrade `void setUrl(Object url)` here?

Reply:
It would be hard since there is no fileResolver available. It would be cumbersome to reroute the call, and I suppose this interface is only used by plugins.
I decided it was not worth migrating.